src/main/java/org/broadinstitute/dsde/consent/ontology/resources/OntologySearchResource.java

```diff
@@ -24,20 +24,26 @@ public OntologySearchResource(AutocompleteAPI api) {
     @GET
     @Produces("application/json")
     public Response getOntologyById(@QueryParam("id") @DefaultValue("") String queryTerm) throws IOException {
-        if (!queryTerm.isEmpty()) {
+        if (queryTerm.isEmpty()) {
+            return Response
+                .status(Response.Status.BAD_REQUEST)
+                .entity(new ErrorResponse(" Ontology ID term cannot be empty. ", Response.Status.BAD_REQUEST.getStatusCode()))
+                .build();
+        }
+        try {
             List<TermResource> result = api.lookupById(queryTerm);
             if(!result.isEmpty()){
                 return Response.ok().entity(result).build();
             } else {
                 return Response.status(Response.Status.NOT_FOUND)
-                        .entity(new ErrorResponse(" Supplied ID doesn't match any known ontology. ", Response.Status.NOT_FOUND.getStatusCode()))
-                        .build();
+                    .entity(new ErrorResponse(" Supplied ID doesn't match any known ontology term. ", Response.Status.NOT_FOUND.getStatusCode()))
+                    .build();
             }
-        } else {
+        } catch (Exception e) {
             return Response
-                    .status(Response.Status.BAD_REQUEST)
-                    .entity(new ErrorResponse(" Ontology ID term cannot be empty. ", Response.Status.BAD_REQUEST.getStatusCode()))
-                    .build();
+                .status(Response.Status.INTERNAL_SERVER_ERROR)
+                .entity(new ErrorResponse(e.getMessage(), Response.Status.INTERNAL_SERVER_ERROR.getStatusCode()))
+                .build();
```

Comment:
**Note to reviewer** This returns a better default error message. Dropwizard otherwise sends a generic `Error handling a request: 7e1ff6255c962657` on `500`'s that we need to go look up in the logs to make any sense of.